### DISCUSSION TITLE
Consider decomposing the driver a bit more

### DISCUSSION THREAD

**Discussion by shueybubbles** [dotnet/SqlClient#1715] [Ideas]
Dependency management when using M.D.S is getting more complicated as time goes by (see #1714 as an example). In particular, MSAL is evolving very quickly. 
Applications that want to implement their own AAD authentication provider are still burdened with M.D.S bringing in its own MSAL dependency tree. 
I'd love to see the default MSAL implementation in M.D.S separated out into its own nuget package that applications could consume if they don't want to create their own. Then applications like SSMS that need to implement MSAL auth with fully custom behaviors and evolve at a different pace won't have to deal with M.D.S package dependencies on Azure components at build time.

**Comment by ErikEJ** [comment]
Duplicate of #1108 ?

**Reply by shueybubbles** [reply]
Yep!
Thx
